pkg/ingestor/ingestcli/client.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ingestcli
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/pkg/util/intest"
+	"golang.org/x/sync/errgroup"
+)
+
+var _ WriteClient = &writeClient{}
+
+type writeClient struct {
+	tikvWorkerURL string
+	clusterID     uint64
+	taskID        int64
+	httpClient    *http.Client
+	commitTS      uint64
+
+	eg          errgroup.Group
+	inited      bool
+	errCh       chan error
+	writer      *io.PipeWriter
+	reader      *io.PipeReader
+	sstFilePath string
+}
+
+// newWriteClient creates a writeClient. It is caller's responsibility to call Close() when it meets an error.
+func newWriteClient(tikvWorkerURL string, clusterID uint64, taskID int64, httpClient *http.Client) *writeClient {
+	return &writeClient{
+		tikvWorkerURL: tikvWorkerURL,
+		clusterID:     clusterID,
+		taskID:        taskID,
+		httpClient:    httpClient,
+	}
+}
+
+func (w *writeClient) init(ctx context.Context) error {
+	intest.Assert(!w.inited)
+	defer func() {
+		w.inited = true
+	}()
+	pr, pw := io.Pipe()
+	url := fmt.Sprintf("%s/write_sst?cluster_id=%d&commit_ts=%d",
+		w.tikvWorkerURL, w.clusterID, w.commitTS)
```

Comment:
there is no need to pass host:port part inside uri, the `httpClient` already have it